include/libtorrent/kademlia/dht_storage.hpp

```diff
@@ -64,6 +70,50 @@ namespace dht
 		void reset();
 	};
 
+	struct TORRENT_EXPORT dht_immutable_item
+	{
+		dht_immutable_item() = default;
+		dht_immutable_item(dht_immutable_item const& item);
+		dht_immutable_item(dht_immutable_item&&) = default;
+		dht_immutable_item& operator=(dht_immutable_item const& item);
+		dht_immutable_item& operator=(dht_immutable_item&&) = default;
+
+		// the actual value
+		std::unique_ptr<char[]> value;
+		// this counts the number of IPs we have seen
+		// announcing this item, this is used to determine
+		// popularity if we reach the limit of items to store
+		bloom_filter<128> ips;
+		// the last time we heard about this item
+		// the correct interpretation of this field
+		// requires a time reference
+		time_point last_seen;
+		// number of IPs in the bloom filter
+		int num_announcers = 0;
+		// size of malloced space pointed to by value
+		int size = 0;
+	};
+
+	struct TORRENT_EXPORT dht_mutable_item : dht_immutable_item
+	{
+		signature sig;
+		sequence_number seq;
+		public_key key;
+		std::string salt;
+	};
+
+	// This structure hold the DHT items (immutable and mutable)
+	// exported by the DHT's storage.
+	struct TORRENT_EXPORT dht_storage_items
+	{
+		// time reference to interpret the item's ``last_seen``
+		// actual last_seen time is (timestamp - total_seconds(last_seen))
```

Comment:
is this comment right? ``total_seconds()`` take a ``duration``, not a ``time_point``, doesn't it?
I take it you mean ``total_seconds(last_seen.time_since_epoch())`` or something like that.